### PR TITLE
Refactor expiring thread code

### PR DESCRIPTION
**Pull request by erksmit**
Moves code related to expiring threads out of the plugin store cog. Instead using a seperate cog, so that any cog can use them.

### CODE REVIEW

File: bot/cogs/utils/api.py [hunk 1]
```diff
@@ -43,7 +44,7 @@ def suggestions(self) -> AgnosticCollection:
         return self.db["Suggestions"]
 
     @property
-    def plugin_threads(self) -> AgnosticCollection:
+    def expiring_threads(self) -> AgnosticCollection:
         """Returns a MongoDB collection for plugin showcase threads."""
         return self.db["Plugin_threads"]
```
Comment (JustAnyones) [inline bot/cogs/utils/api.py:49]: Collection string should be changed

File: bot/cogs/utils/api.py [hunk 2]
```diff
@@ -179,23 +180,28 @@ async def fetch_plugin_by_id(self, plugin_id: int, show_hidden: bool = False) ->
...
             "expires": expire_timestamp
         })
 
-    async def get_archived_plugin_threads(self, timestamp: float) -> List[dict]:
-        """Returns a list of active plugin threads which require archiving."""
-        cursor = self.plugin_threads.find(
+    async def create_expiring_thread(self, message: Message, name: str, expire_timestamp: int):
+        """Creates a new expiring thread"""
+        thread = await message.create_thread(name=name, auto_archive_duration=60)
+        await self.create_new_expiring_thread(thread.id, expire_timestamp)
```
Comment (JustAnyones) [inline bot/cogs/utils/api.py:193]: I would rather have this defined in the Pidroid object

**Review by JustAnyones — CHANGES_REQUESTED** [review]

File: bot/client.py [hunk 3]
```diff
@@ -183,6 +183,11 @@ def _update_guild_configuration(self, guild_id: int, config: GuildConfiguration)
     def _remove_guild_configuration(self, guild_id: int) -> None:
         self.guild_configurations.pop(guild_id)
 
+    async def create_expiring_thread(self, message: Message, name: str, expire_timestamp: int):
```
Comment (JustAnyones) [inline bot/client.py:186]: Add support for specifying custom auto archive duration

**Review by JustAnyones — CHANGES_REQUESTED** [review]

**Review by JustAnyones — CHANGES_REQUESTED** [review]
auto_archive_duration parameter is not type-hinted

**Comment by erksmit** [comment]
My pr is in shambles

**Comment by JustAnyones** [comment]
Looks good, resolve the conflict and it will be merged.

**Review by JustAnyones — APPROVED** [review]

**Pull request merged**